docs/sofapython3/tutorials/Tripod/details/step5.py

```diff
@@ -20,45 +16,37 @@ def __init__(self, *args, **kwargs):
     scene.addMainHeader()
-    scene.addObject('AttachBodyButtonSetting', stiffness=10)  # Set mouse spring stiffness
     scene.addObject('DefaultVisualManagerLoop')
     scene.addObject('FreeMotionAnimationLoop')
     scene.addObject('GenericConstraintSolver', maxIterations=50, tolerance=1e-5)
     scene.Simulation.addObject('GenericConstraintCorrection')
-
+    scene.Simulation.TimeIntegrationSchema.rayleighStiffness = 0.005
+    scene.Settings.mouseButton.stiffness = 10
     scene.VisualStyle.displayFlags = "showBehavior"
 
-    tripod = scene.Modelling.addChild(Tripod())
-
-    scene.addObject(TripodController(name="TripodController",actuators=[tripod.ActuatedArm0, tripod.ActuatedArm1, tripod.ActuatedArm2]))
-
-    scene.Simulation.addChild(tripod.RigidifiedStructure)
-
-    motors = scene.Simulation.addChild("Motors")
-    for i in range(3):
-        motors.addChild(tripod.getChild('ActuatedArm'+str(i)))
+    tripod = Tripod()
+    scene.Modelling.addChild(tripod)
+    scene.Modelling.addObject(TripodController(name="TripodController",actuators=[tripod.ActuatedArm0, tripod.ActuatedArm1, tripod.ActuatedArm2]))
+    scene.Simulation.addChild(tripod)
 
     # Temporary additions to have the system correctly built in SOFA
     # Will no longer be required in SOFA v22.06
     scene.Simulation.addObject('MechanicalMatrixMapper',
-                                 name="mmmFreeCenter",
+                                 name="deformable-freecenter-coupling",
                                  template='Vec3,Rigid3',
-                                 object1="@RigidifiedStructure/DeformableParts/dofs",
-                                 object2="@RigidifiedStructure/FreeCenter/dofs",
-                                 nodeToParse="@RigidifiedStructure/DeformableParts/ElasticMaterialObject")
+                                 object1=tripod.RigidifiedStructure.DeformableParts.dofs.getLinkPath(),
+                                 object2=tripod.RigidifiedStructure.FreeCenter.dofs.getLinkPath(),
+                                 nodeToParse=tripod.RigidifiedStructure.DeformableParts.MechanicalModel.getLinkPath())
 
     for i in range(3):
         scene.Simulation.addObject('MechanicalMatrixMapper',
-                                   name="mmmDeformableAndArm" + str(i),
+                                   name="deformable-arm{i}-coupling".format(i=i),
                                    template='Vec1,Vec3',
-                                   object1="@Modelling/Tripod/ActuatedArm" + str(i) + "/ServoMotor/Articulation/dofs",
-                                   object2="@Simulation/RigidifiedStructure/DeformableParts/dofs",
+                                   object1=tripod["ActuatedArm" + str(i) + ".ServoMotor.Articulation.dofs"].getLinkPath(),
+                                   object2=tripod["RigidifiedStructure.DeformableParts.dofs"].getLinkPath(),
                                    skipJ2tKJ2=True,
```

Comment:
Because it is not possible with the tripod.RigidifiedStructure.FreeCenter.dofs.getLinkPath() syntax to have stuff like "ActuatedArm"+str(i).
This is why the [] was added. But this is not ideal either and we should probably do things like
```python3
for arm in tripod.actuatedArms:
     object2 = arm.getLinkPath()
```